sharding/nodesSetup.go

```diff
@@ -389,12 +499,39 @@ func (ns *NodesSetup) GetRoundDuration() uint64 {
 
 // GetShardConsensusGroupSize returns the shard consensus group size
 func (ns *NodesSetup) GetShardConsensusGroupSize() uint32 {
-	return ns.ConsensusGroupSize
+	ns.mutConfiguration.RLock()
+	defer ns.mutConfiguration.RUnlock()
+
+	return ns.currentShardConsensus.ConsensusGroupSize
 }
 
 // GetMetaConsensusGroupSize returns the metachain consensus group size
```

Comment:
maybe in a future PR, we will extract these chain configurations by epoch in a new component? As they are unrelated to the nodeSetup struct.

Reply:
already done in this PR